### PR TITLE
Added tflite frontend support for quantized mean

### PR DESCRIPTION
**Pull request by tristan-arm**

### CODE REVIEW

**Comment by tqchen** [comment]
cc @FrozenGene @srkreddy1238 @anijain2305 can you take a look?

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -659,7 +659,23 @@ def _convert_reduce(self, relay_op, op):
         reduce_options.Init(op_options.Bytes, op_options.Pos)
         keep_dims = reduce_options.KeepDims()
 
+        if input_tensor.qnn_params:
+            in_expr = _qnn.op.dequantize(data=in_expr,
```
Comment (FrozenGene) [inline python/tvm/relay/frontend/tflite.py]: Hmm...do we have to dequantize -> compute fp32 -> requantize? could we do only integer computation?

File: tests/python/frontend/tflite/test_forward.py [hunk 2]
```diff
@@ -710,16 +710,23 @@ def test_forward_zeros_like():
 # Reduce
 # ------
 
-def _test_reduce(math_op, data, keep_dims=None):
+def _test_reduce(math_op, data, keep_dims=None, quantized=False):
```
Comment (FrozenGene) [inline tests/python/frontend/tflite/test_forward.py]: Suggest breaking into two functions. Another is _test_reduce_quantize

File: python/tvm/relay/frontend/tflite.py (hunk 1 above)
Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py]: Similar concern as that of @FrozenGene 
From maths size, it looks safe to me, but let me know if it wrong

~~~~
Initial Quantized tensor - QA - shape (N, dtype='int8') 

Mean would be 

scale_output * (Qout - zp_out) = scale_in * [(QA[0] + QA[1] .... + QA[N-1]) - N* zp_in]/N

scale_output * (Qout - zp_out) = scale_in * (Mean(QA) - zp_in)
~~~~

So, basically, upcast the quantized tensor to int32, call Int Mean, and then requantize if the output scale/zp are different.

File: tests/python/frontend/tflite/test_forward.py (hunk 2 above)
Comment (tristan-arm) [reply tests/python/frontend/tflite/test_forward.py]: > Suggest breaking into two functions. Another is _test_reduce_quantize

Done

File: python/tvm/relay/frontend/tflite.py (hunk 1 above)
Comment (tristan-arm) [reply python/tvm/relay/frontend/tflite.py]: Updated as suggested above

**Comment by tristan-arm** [comment]
CI failure is unrelated to this patch - I'll kick it off again once the issue is resolved

**Comment by zhiics** [comment]
@tristan-arm CI is consistently failing. Shouldn't relate to your change.

**Comment by zhiics** [comment]
Thanks @tristan-arm @anijain2305 @FrozenGene

**Pull request merged**